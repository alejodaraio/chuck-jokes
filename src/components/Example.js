import React, { Fragment } from "react";

export default ({ value }) => {
  <Fragment>{value}</Fragment>;
};
