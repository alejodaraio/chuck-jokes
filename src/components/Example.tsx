import React from 'react'
import { IStringValue } from '../@types/types.d.stringValue'

const Example: React.FC<IStringValue> = ({ value }) => {
  return <>{value}</>
}
export default Example
