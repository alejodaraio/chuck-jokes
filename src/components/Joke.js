import React, { Fragment } from "react";
import PropTypes from "prop-types";

const Joke = ({ value }) => {
  return (
    <Fragment>
      <h1>Chuck says:</h1>
      {value}
    </Fragment>
  );
};

Joke.propTypes = {
  value: PropTypes.string,
};

export default Joke;
