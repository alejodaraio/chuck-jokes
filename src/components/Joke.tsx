import React, { Fragment } from 'react'
import { IStringValue } from '../@types/types.d.stringValue'

const Joke: React.FC<IStringValue> = ({ value }) => {
  return (
    <Fragment>
      <h1>Chuck says:</h1>
      {value}
    </Fragment>
  )
}

export default Joke
