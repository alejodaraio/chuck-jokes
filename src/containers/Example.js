import React from "react";
import { useExampleContext } from "../contexts/ExampleContext";

export default () => {
  const { currentText } = useExampleContext();

  return <h3>{currentText}</h3>;
};
