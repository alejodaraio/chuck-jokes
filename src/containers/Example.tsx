import React from 'react'
import { ExampleContextType } from '../@types/types.d.example'
import { useExampleContext } from '../contexts/ExampleContext'

export default () => {
  const { currentText } = useExampleContext() as ExampleContextType
  return <h3>{currentText}</h3>
}
