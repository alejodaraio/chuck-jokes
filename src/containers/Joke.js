import React, { useEffect } from "react";
import { useJoke } from "../contexts/JokeContext";
import Joke from "../components/Joke";

export default () => {
  const [{ joke }, { getJoke }] = useJoke();

  useEffect(() => {
    getJoke();
  }, []);

  return <Joke value={joke} />;
};
