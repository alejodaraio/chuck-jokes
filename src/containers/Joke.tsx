import React, { useEffect } from 'react'
import { useJoke } from '../contexts/JokeContext'
import Joke from '../components/Joke'
import { JokeContextType } from '../@types/types.d.joke'

export default () => {
  const { joke, getJoke } = useJoke() as JokeContextType

  useEffect(() => {
    getJoke()
  }, [])

  return <Joke value={joke} />
}
