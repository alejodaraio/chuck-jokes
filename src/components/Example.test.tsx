import React from 'react'
import renderer from 'react-test-renderer'
import Example from '../components/Example'
import { ExampleProvider } from '../contexts/ExampleContext'
import Joke from './Joke'

describe('Example Component test.', () => {
  it('should match with Snapshot.', () => {
    const mockValue = 'This is an example test'
    const component = renderer.create(<Example value={mockValue} />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
