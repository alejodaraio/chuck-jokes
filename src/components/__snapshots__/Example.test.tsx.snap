// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example Component test. should match with Snapshot. 1`] = `"This is an example test"`;
